1.19.0-alpha.0

matteobruni opened the pull request.

dependabot[bot] committed:
build(deps-dev): bump @types/react from 16.9.51 to 16.9.52

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 16.9.51 to 16.9.52.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump ts-loader from 8.0.4 to 8.0.5

Bumps [ts-loader](https://github.com/TypeStrong/ts-loader) from 8.0.4 to 8.0.5.
- [Release notes](https://github.com/TypeStrong/ts-loader/releases)
- [Changelog](https://github.com/TypeStrong/ts-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/TypeStrong/ts-loader/compare/8.0.4...v8.0.5)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #955 from matteobruni/dependabot/npm_and_yarn/dev/types/react-16.9.52

build(deps-dev): bump @types/react from 16.9.51 to 16.9.52

matteobruni committed:
Merge pull request #956 from matteobruni/dependabot/npm_and_yarn/dev/ts-loader-8.0.5

build(deps-dev): bump ts-loader from 8.0.4 to 8.0.5

matteobruni committed:
build: updated angular

dBiazioli committed:
Added README translations to Brazilian Portuguese

matteobruni committed:
Merge pull request #957 from dBiazioli/master

Added README translations to Brazilian Portuguese

matteobruni committed:
Merge branch 'master' into dev

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.28.2 to 2.31.0

Bumps [rollup](https://github.com/rollup/rollup) from 2.28.2 to 2.31.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.28.2...v2.31.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @vue/eslint-config-typescript from 5.1.0 to 7.0.0

Bumps [@vue/eslint-config-typescript](https://github.com/vuejs/eslint-config-typescript) from 5.1.0 to 7.0.0.
- [Release notes](https://github.com/vuejs/eslint-config-typescript/releases)
- [Commits](https://github.com/vuejs/eslint-config-typescript/compare/v5.1.0...v7.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #959 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.31.0

build(deps-dev): bump rollup from 2.28.2 to 2.31.0

matteobruni committed:
Merge pull request #960 from matteobruni/dependabot/npm_and_yarn/dev/vue/eslint-config-typescript-7.0.0

build(deps-dev): bump @vue/eslint-config-typescript from 5.1.0 to 7.0.0

matteobruni committed:
build: try catch on install script

matteobruni committed:
build: updated some readmes

matteobruni committed:
fix: fix an import and a return

matteobruni committed:
build: prettified code

dependabot[bot] committed:
build(deps-dev): bump @babel/preset-env from 7.11.5 to 7.12.1

Bumps [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env) from 7.11.5 to 7.12.1.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.1/packages/babel-preset-env)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @types/chai from 4.2.13 to 4.2.14

Bumps [@types/chai](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/chai) from 4.2.13 to 4.2.14.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/chai)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #963 from matteobruni/dependabot/npm_and_yarn/dev/babel/preset-env-7.12.1

build(deps-dev): bump @babel/preset-env from 7.11.5 to 7.12.1

matteobruni committed:
Merge pull request #964 from matteobruni/dependabot/npm_and_yarn/dev/types/chai-4.2.14

build(deps-dev): bump @types/chai from 4.2.13 to 4.2.14

matteobruni committed:
build: fixes #962

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.5
 - inferno-particles@1.1.5
 - jquery-particles@1.18.5
 - preact-particles@1.18.5
 - react-tsparticles@1.18.5
 - svelte-particles@1.1.5
 - particles.vue@2.1.5
 - particles.vue3@1.1.5
 - tsparticles-editor@1.18.5
 - tsparticles@1.18.5
 - ng-particles-demo@1.3.5
 - editor-demo@1.3.5
 - inferno-particles-demo@1.1.5
 - jquery-particles-demo@1.3.5
 - tsparticles-demo@1.3.5
 - preact-particles-demo@1.3.5
 - react-tsparticles-demo@1.3.5
 - svelte-demo@1.1.5
 - particles.vue-demo@1.3.5
 - particles.vue3-demo@1.1.5
 - tsparticles-preset-60fps@1.3.5
 - tsparticles-preset-background-mask@1.3.5
 - tsparticles-preset-basic@1.3.5
 - tsparticles-preset-big-circles@1.3.5
 - tsparticles-preset-bouncing@1.3.5
 - tsparticles-preset-fire@1.3.5
 - tsparticles-preset-font-awesome@1.3.5
 - tsparticles-preset-fountain@1.3.5
 - tsparticles-preset-snow@1.3.5
 - tsparticles-preset-stars@1.3.5
 - tsparticles-shape-bubble@1.3.5
 - tsparticles-shape-heart@1.3.5
 - tsparticles-shape-multiline-text@1.3.5
 - tsparticles-shape-spiral@1.3.5
 - cra-template-particles-typescript@1.1.5
 - cra-template-particles@1.1.5

matteobruni committed:
Merge pull request #961 from matteobruni/dev

1.18.5

matteobruni committed:
build: added webpack 5 to core/main

matteobruni committed:
build: added webpack5 to editor

matteobruni committed:
build: added webpack5 to spiral shape

matteobruni committed:
build: added webpack5 to bubble shape

matteobruni committed:
build: added webpack5 to heart shape

matteobruni committed:
build: added webpack5 to multiline text shape, and removed tslib helpers from shapes

matteobruni committed:
build: added webpack5 to 60fps preset

matteobruni committed:
build: added webpack5 to backgroundMask preset

matteobruni committed:
build: added webpack5 to basic preset

matteobruni committed:
build: added webpack5 to big circles preset

matteobruni committed:
build: added webpack5 to bouncing preset

matteobruni committed:
build: added webpack5 to fire preset

matteobruni committed:
build: added webpack5 to font awesome preset

matteobruni committed:
build: added webpack5 to fountain preset

matteobruni committed:
build: added webpack5 to snow preset

matteobruni committed:
build: added webpack5 to stars preset

matteobruni committed:
build: updated yarn.lock

matteobruni committed:
build: added webpack5 to inferno component

matteobruni committed:
build: added webpack5 to jQuery component

matteobruni committed:
build: updated types/node on jquery component

matteobruni committed:
build: added webpack5 to Preact component

matteobruni committed:
build: updated types/node on preact component

matteobruni committed:
build: added webpack5 to React component

matteobruni committed:
build: completed webpack5 transition, inferno demo is broken for now (can't start)

matteobruni committed:
build: fixed inferno demo

matteobruni committed:
build: fixed build for webpack5

matteobruni committed:
Merge pull request #965 from matteobruni/webpack5

Migration to Webpack 5.x

matteobruni committed:
build: removed debug console.log from react component

matteobruni committed:
fix: fixed install script for Node 8.x, fixes #966

dependabot[bot] committed:
build(deps-dev): bump rollup-plugin-svelte from 6.0.1 to 6.0.2

Bumps [rollup-plugin-svelte](https://github.com/sveltejs/rollup-plugin-svelte) from 6.0.1 to 6.0.2.
- [Release notes](https://github.com/sveltejs/rollup-plugin-svelte/releases)
- [Changelog](https://github.com/sveltejs/rollup-plugin-svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/rollup-plugin-svelte/compare/v6.0.1...v6.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #968 from matteobruni/dependabot/npm_and_yarn/dev/rollup-plugin-svelte-6.0.2

build(deps-dev): bump rollup-plugin-svelte from 6.0.1 to 6.0.2

matteobruni committed:
build: removed autogenerated index.html

matteobruni committed:
build: updated gitignore

matteobruni committed:
build: added some presets demos

dependabot[bot] committed:
build(deps-dev): bump ts-loader from 8.0.5 to 8.0.6

Bumps [ts-loader](https://github.com/TypeStrong/ts-loader) from 8.0.5 to 8.0.6.
- [Release notes](https://github.com/TypeStrong/ts-loader/releases)
- [Changelog](https://github.com/TypeStrong/ts-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/TypeStrong/ts-loader/compare/v8.0.5...v8.0.6)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @vue/compiler-sfc from 3.0.1 to 3.0.2

Bumps [@vue/compiler-sfc](https://github.com/vuejs/vue-next/tree/HEAD/packages/compiler-sfc) from 3.0.1 to 3.0.2.
- [Release notes](https://github.com/vuejs/vue-next/releases)
- [Changelog](https://github.com/vuejs/vue-next/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vue-next/commits/v3.0.2/packages/compiler-sfc)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #969 from matteobruni/dependabot/npm_and_yarn/dev/ts-loader-8.0.6

build(deps-dev): bump ts-loader from 8.0.5 to 8.0.6

matteobruni committed:
Merge pull request #970 from matteobruni/dependabot/npm_and_yarn/dev/vue/compiler-sfc-3.0.2

build(deps-dev): bump @vue/compiler-sfc from 3.0.1 to 3.0.2

matteobruni committed:
build: fixed demo samples for all presets and shapes in the workspace

matteobruni committed:
Update nodejs.yml

matteobruni committed:
Update nodejs.yml

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.6
 - inferno-particles@1.1.6
 - jquery-particles@1.18.6
 - preact-particles@1.18.6
 - react-tsparticles@1.18.6
 - svelte-particles@1.1.6
 - particles.vue@2.1.6
 - particles.vue3@1.1.6
 - tsparticles-editor@1.18.6
 - tsparticles@1.18.6
 - ng-particles-demo@1.3.6
 - editor-demo@1.3.6
 - inferno-particles-demo@1.1.6
 - jquery-particles-demo@1.3.6
 - tsparticles-demo@1.3.6
 - preact-particles-demo@1.3.6
 - react-tsparticles-demo@1.3.6
 - svelte-demo@1.1.6
 - particles.vue-demo@1.3.6
 - particles.vue3-demo@1.1.6
 - tsparticles-preset-60fps@1.3.6
 - tsparticles-preset-background-mask@1.3.6
 - tsparticles-preset-basic@1.3.6
 - tsparticles-preset-big-circles@1.3.6
 - tsparticles-preset-bouncing@1.3.6
 - tsparticles-preset-fire@1.3.6
 - tsparticles-preset-font-awesome@1.3.6
 - tsparticles-preset-fountain@1.3.6
 - tsparticles-preset-snow@1.3.6
 - tsparticles-preset-stars@1.3.6
 - tsparticles-shape-bubble@1.3.6
 - tsparticles-shape-heart@1.3.6
 - tsparticles-shape-multiline-text@1.3.6
 - tsparticles-shape-spiral@1.3.6
 - cra-template-particles-typescript@1.1.6
 - cra-template-particles@1.1.6

matteobruni committed:
Merge pull request #967 from matteobruni/dev

1.18.6

matteobruni committed:
build: updated deps

matteobruni committed:
build: updated deps

matteobruni committed:
Merge branch 'master' into dev

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.1.2 to 8.1.3

Bumps [postcss](https://github.com/postcss/postcss) from 8.1.2 to 8.1.3.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/master/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.1.2...8.1.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump sass-loader from 10.0.3 to 10.0.4

Bumps [sass-loader](https://github.com/webpack-contrib/sass-loader) from 10.0.3 to 10.0.4.
- [Release notes](https://github.com/webpack-contrib/sass-loader/releases)
- [Changelog](https://github.com/webpack-contrib/sass-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/sass-loader/compare/v10.0.3...v10.0.4)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #974 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.1.3

build(deps-dev): bump postcss from 8.1.2 to 8.1.3

matteobruni committed:
Merge pull request #975 from matteobruni/dependabot/npm_and_yarn/dev/sass-loader-10.0.4

build(deps-dev): bump sass-loader from 10.0.3 to 10.0.4

matteobruni committed:
build: updated webpack

Commit:
test(main): add test for getParticleBaseVelocity

matteobruni committed:
Merge pull request #976 from mendaomn/896-add-get-particle-base-velocity-test

test(main): add test for getParticleBaseVelocity

matteobruni committed:
build: migrated from yarn to pnpm and from lerna to rush

matteobruni committed:
build: migrated from yarn to pnpm and from lerna to rush

matteobruni committed:
build: updated ci script

matteobruni committed:
build: updated ci script

matteobruni committed:
build: ci updated

matteobruni committed:
build: update ci script

matteobruni committed:
build: update ci script

matteobruni committed:
build: ci updated

matteobruni committed:
build: fixed package.json for pug command

matteobruni committed:
build: removed build from simple demos

matteobruni committed:
build: added a fake build command where not needed

matteobruni committed:
build: build script can have warnings

matteobruni committed:
build: updated contributing file

matteobruni committed:
build: updated demos

matteobruni committed:
build: updated rush config and added a deprecate script

matteobruni committed:
build: working on version policies

matteobruni committed:
fix: absorbers and emitters can have partial positions, the other coordinate will be randomized

matteobruni committed:
fix: absobers and emitters plugin now will add some custom methods to Container

matteobruni committed:
build: fixed some imports

matteobruni committed:
build: install script supports next and nuxt too

matteobruni committed:
build: little refactoring

matteobruni committed:
build: moved private static to unexported file functions/variables

matteobruni committed:
build: updated some demo presets

Commit:
test(main): add test for loadImage

matteobruni committed:
build: changed the build script from prepare to prepack

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.7
 - inferno-particles@1.1.7
 - jquery-particles@1.18.7
 - preact-particles@1.18.7
 - react-tsparticles@1.18.7
 - svelte-particles@1.1.7
 - particles.vue@2.1.7
 - particles.vue3@1.1.7
 - tsparticles-editor@1.18.7
 - tsparticles@1.18.7
 - ng-particles-demo@1.3.7
 - editor-demo@1.3.7
 - inferno-particles-demo@1.1.7
 - jquery-particles-demo@1.3.7
 - tsparticles-demo@1.3.7
 - preact-particles-demo@1.3.7
 - react-tsparticles-demo@1.3.7
 - svelte-demo@1.1.7
 - particles.vue-demo@1.3.7
 - particles.vue3-demo@1.1.7
 - tsparticles-preset-60fps@1.3.7
 - tsparticles-preset-background-mask@1.3.7
 - tsparticles-preset-basic@1.3.7
 - tsparticles-preset-big-circles@1.3.7
 - tsparticles-preset-bouncing@1.3.7
 - tsparticles-preset-fire@1.3.7
 - tsparticles-preset-font-awesome@1.3.7
 - tsparticles-preset-fountain@1.3.7
 - tsparticles-preset-snow@1.3.7
 - tsparticles-preset-stars@1.3.7
 - tsparticles-shape-bubble@1.3.7
 - tsparticles-shape-heart@1.3.7
 - tsparticles-shape-multiline-text@1.3.7
 - tsparticles-shape-spiral@1.3.7
 - cra-template-particles-typescript@1.1.7
 - cra-template-particles@1.1.7

matteobruni committed:
Merge pull request #978 from matteobruni/dev

1.18.7

matteobruni committed:
Merge branch 'master' into pnpm

matteobruni committed:
Merge pull request #980 from mendaomn/896-add-load-image-test

test(main): add test for loadImage

matteobruni committed:
build: updates

matteobruni committed:
build: updated pnpm yaml

matteobruni committed:
Merge pull request #977 from matteobruni/pnpm

Pnpm/Rush instead of Yarn/Lerna

matteobruni committed:
build: some refactoring in container particles object
feat: added factor options to repulse and attract options objects

ghost commented:
## DeepCode failed to analyze this pull request
Something went wrong despite trying multiple times, sorry about that.
Please comment this pull request with "Retry DeepCode" to manually retry, or [contact us](https://www.deepcode.ai/feedback?select=4&utm_source=gh_review) so that a human can look into the issue.

Commit:
test(main): add test for getAvailablePlugins

matteobruni committed:
Merge pull request #983 from mendaomn/896-add-plugins-test

test(main): add test for getAvailablePlugins

matteobruni committed:
feat: added particles groups to override options, missing docs

matteobruni committed:
docs: updated tsparticles docs

matteobruni committed:
build: updated react to the latest version (17, but kept support for 16)

matteobruni committed:
build: fixed comment style

Commit:
feat: add static orbit

matteobruni committed:
build: updated typescript version and some dependencies

Commit:
refactor: refactor orbit code

Commit:
refactor: refactor orbit code

Commit:
test(main): add test for downloadSvgImage

matteobruni committed:
Merge pull request #987 from mendaomn/896-add-download-svg-image-test

test(main): add test for downloadSvgImage

Commit:
refactor: refactor orbit code

Commit:
feat: add orbit to demo

IngiRodr committed:
add spanish translate readme

matteobruni committed:
Merge pull request #989 from IngiRodr/IngiRodr-add-spanish-translate

add spanish translate readme

matteobruni committed:
build: code prettiefied

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
build: various updates and added pre-commit hook

matteobruni committed:
build: removed rush update from pre-commit

Commit:
fix: orbit width radius rotation

Commit:
refactor: orbit width

Commit:
refactor: orbit width

Commit:
refactor: orbit

Commit:
refactor: orbit

Commit:
refactor: orbit

Commit:
refactor: orbit

matteobruni committed:
build: update angular component

matteobruni committed:
build: restored rush update, it's better have it

matteobruni committed:
feat: first working implementation of zIndex for particles

matteobruni commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 147
           

Complexity increasing per file
==============================
- common/scripts/install-run.js  7
- scripts/deprecate-sh.js  2
- core/main/src/Interactions/Particles/Linker.ts  17
- core/main/src/Interactions/External/Grabber.ts  10
- core/main/src/Options/Classes/Particles/Particles.ts  5
- core/main/src/Options/Classes/Options.ts  1
- components/preact/src/Particles.tsx  5
- core/main/src/Options/Classes/ManualParticle.ts  6
- core/main/src/Utils/CanvasUtils.ts  13
- core/main/src/Interactions/External/TrailMaker.ts  7
- core/main/src/Core/Particles.ts  13
- common/scripts/install-run-rush.js  7
- core/main/src/Interactions/External/Lighter.ts  4
- core/main/src/Interactions/External/Bubbler.ts  15
- core/main/src/Options/Classes/Interactivity/Modes/RepulseBase.ts  1
- core/main/src/Interactions/External/Connector.ts  8
- core/main/src/Interactions/External/Repulser.ts  10
- components/inferno/src/Particles.tsx  5
- core/main/src/Interactions/External/Bouncer.ts  4
- core/main/src/Core/Retina.ts  12
- core/main/src/Options/Classes/Interactivity/Modes/Attract.ts  1
- core/main/src/Core/Canvas.ts  34
- core/main/src/Interactions/External/Attractor.ts  7
- common/config/rush/pnpmfile.js  1
- components/react/src/Particles.tsx  5
         

Clones added
============
- common/scripts/install-run.js  1
- core/main/src/Core/Container.ts  2
- common/scripts/install-run-rush.js  1
         

Clones removed
==============
+ core/main/src/Utils/CanvasUtils.ts  -2
+ core/main/src/Interactions/External/TrailMaker.ts  -1
+ core/main/src/Options/Classes/Interactivity/Modes/RepulseBase.ts  -1
+ core/main/src/Options/Classes/Interactivity/Modes/Attract.ts  -1
+ core/main/src/Interactions/External/Attractor.ts  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531&bid=20743896)

matteobruni commented on common/scripts/install-run-rush.js:
```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See the @microsoft/rush package's LICENSE file for license information.
+var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
+    if (k2 === undefined) k2 = k;
```
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni replied:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on core/main/src/Utils/ColorUtils.ts:
```diff
@@ -5,6 +5,96 @@ import type { IImage } from "../Core/Interfaces/IImage";
+    return p;
+}
+
+function stringToRgba(input: string): IRgba | undefined {
```
Codacy found an issue: [non-arrow functions are forbidden](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+        if (fs.existsSync(sourceNpmrcPath)) {
+            _copyAndTrimNpmrcFile(sourceNpmrcPath, targetNpmrcPath);
+        }
+        else if (fs.existsSync(targetNpmrcPath)) {
```
Codacy found an issue: [Found fs.existsSync with non literal argument at index 0](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+ */
+function _syncNpmrc(sourceNpmrcFolder, targetNpmrcFolder, useNpmrcPublish) {
+    const sourceNpmrcPath = path.join(sourceNpmrcFolder, !useNpmrcPublish ? '.npmrc' : '.npmrc-publish');
+    const targetNpmrcPath = path.join(targetNpmrcFolder, '.npmrc');
```
Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/inferno/traduction/README.de.md:
```diff
@@ -0,0 +1,193 @@
+| canvasClassName | string | Der Klassenname des canvas. |
+
+| container | object | Die Instanz des [Partikel-Containers]
+(https://github.com/matteobruni/tsparticles/wiki/Particles-Container-class) |
```
Codacy found an issue: [[no-literal-urls] Don’t use literal URLs without angle brackets](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run-rush.js:
```diff
@@ -0,0 +1,86 @@
+const path = __importStar(require("path"));
+const fs = __importStar(require("fs"));
+const install_run_1 = require("./install-run");
+const PACKAGE_NAME = '@microsoft/rush';
```
Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See the @microsoft/rush package's LICENSE file for license information.
+var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
+    if (k2 === undefined) k2 = k;
```
Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+ *
+ * IMPORTANT: THIS CODE SHOULD BE KEPT UP TO DATE WITH Utilities.copyAndTrimNpmrcFile()
+ */
+function _copyAndTrimNpmrcFile(sourceNpmrcPath, targetNpmrcPath) {
```
Codacy found an issue: [Function '_copyAndTrimNpmrcFile' has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run-rush.js:
```diff
@@ -0,0 +1,86 @@
+    }
+    install_run_1.runWithErrorAndStatusCode(() => {
+        const version = _getRushVersion();
+        console.log(`The rush.json configuration requests Rush version ${version}`);
```
Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/preact/src/Particles.tsx:
```diff
@@ -7,6 +7,9 @@ import type { IParticlesProps } from "./IParticlesProps";
 import { MutableRefObject } from "react";
 
+/**
+ * @param {{id?: string,width?: string,height?: string,options?: ISourceOptions,params?: ISourceOptions,style?: CSSProperties,className?: string,canvasClassName?: string,container?: RefObject<Container>}}
```
Codacy found an issue: [Exceeds maximum line length of 120](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+ * Parse a package specifier (in the form of name\@version) into name and version parts.
+ */
+function _parsePackageSpecifier(rawPackageSpecifier) {
+    rawPackageSpecifier = (rawPackageSpecifier || '').trim();
```
Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/react/traduction/README.pt-br.md:
```diff
@@ -0,0 +1,164 @@
+| style           | object | O estilo do elemento canvas.                                                                                     |
+| className       | string | Nome da classe do elemento que contém o canvas.                                                                                |
+| canvasClassName | string | Nome da classe do canvas.                                                                                        |
+| container       | object | Instância do [particles container](https://github.com/matteobruni/tsparticles/wiki/Particles-Container-class) |
```
Codacy found an issue: [[table-pipe-alignment] Misaligned table fence](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/preact/traduction/README.pt-br.md:
```diff
@@ -0,0 +1,145 @@
+
+| Prop            | Tipo   | Definição                                                                                                           |
+| --------------- | ------ | -------------------------------------------------------------------------------------------------------------------- |
+| width           | string | A largura do canvas.                                                                                             |
```
Codacy found an issue: [[table-pipe-alignment] Misaligned table fence](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/react/traduction/README.pt-br.md:
```diff
@@ -0,0 +1,164 @@
+| --------------- | ------ | -------------------------------------------------------------------------------------------------------------------- |
+| width           | string | A largura do canvas.                                                                                             |
+| height          | string | A altura do canvas.                                                                                            |
+| options         | object | As options das instâncias de partículas.                                                                               |
```
Codacy found an issue: [[table-pipe-alignment] Misaligned table fence](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/config/rush/pnpmfile.js:
```diff
@@ -0,0 +1,38 @@
+ * The `context` parameter provides a log() function.
+ * The return value is the updated object.
+ */
+function readPackage(packageJson, context) {
```
Codacy found an issue: ['context' is defined but never used.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/react/traduction/README.pt-br.md:
```diff
@@ -0,0 +1,164 @@
+| height          | string | A altura do canvas.                                                                                            |
+| options         | object | As options das instâncias de partículas.                                                                               |
+| style           | object | O estilo do elemento canvas.                                                                                     |
+| className       | string | Nome da classe do elemento que contém o canvas.                                                                                |
```
Codacy found an issue: [[table-pipe-alignment] Misaligned table fence](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+/**
+ * Parse a package specifier (in the form of name\@version) into name and version parts.
+ */
+function _parsePackageSpecifier(rawPackageSpecifier) {
```
Codacy found an issue: [Function '_parsePackageSpecifier' has a complexity of 5. Maximum allowed is 4.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/inferno/traduction/README.pt-br.md:
```diff
@@ -0,0 +1,145 @@
+| width           | string | A largura do canvas.                                                                                             |
+| height          | string | A altura do canvas.                                                                                            |
+| options         | object | As options das instâncias de partículas.                                                                               |
+| style           | object | O estilo do elemento canvas.                                                                                     |
```
Codacy found an issue: [[table-pipe-alignment] Misaligned table fence](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
+    Object.defineProperty(o, "default", { enumerable: true, value: v });
+}) : function(o, v) {
+    o["default"] = v;
```
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+                    // Remove the leading "${" and the trailing "}" from the token
+                    const environmentVariableName = token.substring(2, token.length - 1);
+                    // Is the environment variable defined?
+                    if (!process.env[environmentVariableName]) {
```
Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run.js:
```diff
@@ -0,0 +1,471 @@
+ */
+function _copyAndTrimNpmrcFile(sourceNpmrcPath, targetNpmrcPath) {
+    console.log(`Copying ${sourceNpmrcPath} --> ${targetNpmrcPath}`); // Verbose
+    let npmrcFileLines = fs.readFileSync(sourceNpmrcPath).toString().split('\n');
```
Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/react/traduction/README.de.md:
```diff
@@ -0,0 +1,216 @@
+
+- `cra-template-particles`: Einfache ReactJS-Vorlage mit Vollbildschirm-Partikeln, unter Verwendung von JavaScript
+
+- `cra-template-particles-typescript`: Einfache ReactJS-Vorlage mit Vollbildpartikeln, unter Verwendung von TypeScript
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on components/inferno/traduction/README.pt-br.md:
```diff
@@ -0,0 +1,145 @@
+| height          | string | A altura do canvas.                                                                                            |
+| options         | object | As options das instâncias de partículas.                                                                               |
+| style           | object | O estilo do elemento canvas.                                                                                     |
+| className       | string | Nome da classe do elemento que contém o canvas.                                                                                |
```
Codacy found an issue: [[table-pipe-alignment] Misaligned table fence](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni commented on common/scripts/install-run-rush.js:
```diff
@@ -0,0 +1,86 @@
+    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
+}) : (function(o, m, k, k2) {
+    if (k2 === undefined) k2 = k;
+    o[k2] = m[k];
```
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/gh/matteobruni/tsparticles/pullRequest?prid=6397531)

matteobruni committed:
fix: improved mouse events handling

matteobruni committed:
fix: fixed attract mode

matteobruni committed:
fix: fixed particles attract mode

matteobruni committed:
Update Attractor.ts

matteobruni committed:
build: prettiefied code

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.8
 - inferno-particles@1.1.8
 - jquery-particles@1.18.8
 - preact-particles@1.18.8
 - react-tsparticles@1.18.8
 - svelte-particles@1.1.8
 - particles.vue@2.1.8
 - particles.vue3@1.1.8
 - tsparticles-editor@1.18.8
 - tsparticles@1.18.8
 - ng-particles-demo@1.3.8
 - editor-demo@1.3.8
 - inferno-particles-demo@1.1.8
 - jquery-particles-demo@1.3.8
 - tsparticles-demo@1.3.8
 - preact-particles-demo@1.3.8
 - react-tsparticles-demo@1.3.8
 - svelte-demo@1.1.8
 - particles.vue-demo@1.3.8
 - particles.vue3-demo@1.1.8
 - tsparticles-preset-60fps@1.3.8
 - tsparticles-preset-background-mask@1.3.8
 - tsparticles-preset-basic@1.3.8
 - tsparticles-preset-big-circles@1.3.8
 - tsparticles-preset-bouncing@1.3.8
 - tsparticles-preset-fire@1.3.8
 - tsparticles-preset-font-awesome@1.3.8
 - tsparticles-preset-fountain@1.3.8
 - tsparticles-preset-snow@1.3.8
 - tsparticles-preset-stars@1.3.8
 - tsparticles-shape-bubble@1.3.8
 - tsparticles-shape-heart@1.3.8
 - tsparticles-shape-multiline-text@1.3.8
 - tsparticles-shape-spiral@1.3.8
 - cra-template-particles-typescript@1.1.8
 - cra-template-particles@1.1.8

matteobruni committed:
Merge pull request #990 from matteobruni/1.18.8

1.18.8

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
build: updated pnpm lock

matteobruni committed:
fix: attract has its own distance and particle size affects it

matteobruni committed:
build: pnpm lock

matteobruni committed:
build: lockfile

matteobruni committed:
build: used UMD output instead of CJS

matteobruni committed:
build: used UMD output instead of CJS

matteobruni committed:
Update Particles.md

matteobruni committed:
build: updated docs, output too but not published yet since there are no new features

matteobruni committed:
Merge pull request #991 from matteobruni/docs

Docs Fix

matteobruni committed:
build: re-enabled json schema built from typescript

matteobruni committed:
build: removed and ignored schema

matteobruni committed:
build: restored build on commit

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
docs: updated particles md page

Pingviinituutti committed:
feat: caps z-index values at -9999 and 9999. Better scaling of other values depending on z-index

Pingviinituutti committed:
feat: adds options to z-index property

Pingviinituutti committed:
feat: add particle social distancing capabilities

matteobruni committed:
feat: added oct-tree for 3d implementations
build: fixed build without the schema folder

Pingviinituutti committed:
fix: camelcase variables, fix requested changes

matteobruni committed:
feat: added sphere implementation to query 3d ranges
fix: fixed circle query
feat: added 3d position to particle

Pingviinituutti committed:
fix: fix requested changes

matteobruni committed:
Merge pull request #992 from Pingviinituutti/add-z-index-drawing-order

feat: add z-index option to Particles, renders particles in that order

matteobruni committed:
feat: improved zIndex

matteobruni committed:
build: prettiefied code

matteobruni committed:
feat: added groups support to push/remove click modes
feat: collisions now supports zIndex

Pingviinituutti committed:
feat: repulse without collision enabled

matteobruni committed:
fix: links now won't link non-existent particles

matteobruni committed:
build: updated lock file

matteobruni committed:
build: updated lock file

matteobruni committed:
fix: fixes #996
fix: fixes #947

matteobruni committed:
fix: links won't link non-existing particles

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.9
 - inferno-particles@1.1.9
 - jquery-particles@1.18.9
 - preact-particles@1.18.9
 - react-tsparticles@1.18.9
 - svelte-particles@1.1.9
 - particles.vue@2.1.9
 - particles.vue3@1.1.9
 - tsparticles-editor@1.18.9
 - tsparticles@1.18.9
 - ng-particles-demo@1.3.9
 - editor-demo@1.3.9
 - inferno-particles-demo@1.1.9
 - jquery-particles-demo@1.3.9
 - tsparticles-demo@1.3.9
 - preact-particles-demo@1.3.9
 - react-tsparticles-demo@1.3.9
 - svelte-demo@1.1.9
 - particles.vue-demo@1.3.9
 - particles.vue3-demo@1.1.9
 - tsparticles-preset-60fps@1.3.9
 - tsparticles-preset-background-mask@1.3.9
 - tsparticles-preset-basic@1.3.9
 - tsparticles-preset-big-circles@1.3.9
 - tsparticles-preset-bouncing@1.3.9
 - tsparticles-preset-fire@1.3.9
 - tsparticles-preset-font-awesome@1.3.9
 - tsparticles-preset-fountain@1.3.9
 - tsparticles-preset-snow@1.3.9
 - tsparticles-preset-stars@1.3.9
 - tsparticles-shape-bubble@1.3.9
 - tsparticles-shape-heart@1.3.9
 - tsparticles-shape-multiline-text@1.3.9
 - tsparticles-shape-spiral@1.3.9
 - cra-template-particles-typescript@1.1.9
 - cra-template-particles@1.1.9

matteobruni committed:
Merge pull request #997 from matteobruni/r-1.18.9

1.18.9

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
build: updated lockfile

matteobruni committed:
build: updated build configurations

matteobruni committed:
build: updated build configurations

matteobruni committed:
build: fixed build commands

Pingviinituutti committed:
fix: moves particle repulsion to interactions

Pingviinituutti committed:
fix: makes particle repulsion work like mouse repulsion

matteobruni committed:
build: added slim demos to check #999

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.10
 - inferno-particles@1.1.10
 - jquery-particles@1.18.10
 - preact-particles@1.18.10
 - react-tsparticles@1.18.10
 - svelte-particles@1.1.10
 - particles.vue@2.1.10
 - particles.vue3@1.1.10
 - tsparticles-editor@1.18.10
 - tsparticles@1.18.10
 - ng-particles-demo@1.3.10
 - editor-demo@1.3.10
 - inferno-particles-demo@1.1.10
 - jquery-particles-demo@1.3.10
 - tsparticles-demo@1.3.10
 - preact-particles-demo@1.3.10
 - react-tsparticles-demo@1.3.10
 - svelte-demo@1.1.10
 - particles.vue-demo@1.3.10
 - particles.vue3-demo@1.1.10
 - tsparticles-preset-60fps@1.3.10
 - tsparticles-preset-background-mask@1.3.10
 - tsparticles-preset-basic@1.3.10
 - tsparticles-preset-big-circles@1.3.10
 - tsparticles-preset-bouncing@1.3.10
 - tsparticles-preset-fire@1.3.10
 - tsparticles-preset-font-awesome@1.3.10
 - tsparticles-preset-fountain@1.3.10
 - tsparticles-preset-snow@1.3.10
 - tsparticles-preset-stars@1.3.10
 - tsparticles-shape-bubble@1.3.10
 - tsparticles-shape-heart@1.3.10
 - tsparticles-shape-multiline-text@1.3.10
 - tsparticles-shape-spiral@1.3.10
 - cra-template-particles-typescript@1.1.10
 - cra-template-particles@1.1.10

matteobruni committed:
Merge pull request #998 from matteobruni/r-1.18.10

1.18.10

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
build: fixed build files

Pingviinituutti committed:
fix: removes unnecessary definitions
fix: moves Repulser before Linker

Pingviinituutti committed:
Merge remote-tracking branch 'upstream/dev' into add-repulse-to-particles

matteobruni committed:
Merge pull request #993 from Pingviinituutti/add-repulse-to-particles

feat: add repulse to particles

matteobruni committed:
docs: updated GH templates

matteobruni committed:
Update bug_report.md

matteobruni committed:
Update custom.md

matteobruni committed:
Update feature_request.md

matteobruni committed:
Update custom.md

matteobruni committed:
build: sorted properties

matteobruni committed:
Merge pull request #1000 from matteobruni/gh-templates

GitHub templates

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
build: updates

matteobruni committed:
build: code prettiefied

matteobruni committed:
build: updated lockfile

matteobruni committed:
build: updated lock file

matteobruni committed:
build: removed build on commit, too much time wasting

matteobruni committed:
fix: size and opacity are now clamped, this should fix #1001

matteobruni committed:
fix: opacity and size clamped, this should fix #1001

matteobruni committed:
feat: opacity animation options have now startValue and destroy like size animation options

matteobruni committed:
refactor: simplified destroy management on Updater.ts

matteobruni committed:
refactor: splitted some big methods

matteobruni committed:
fix: fixed preact definitions to fix #1003

matteobruni committed:
fix: mouse events works a lot better when interactivity element is not canvas

matteobruni committed:
fix: fixed parent interactivity mode

matteobruni committed:
feat: added new opacity options to editor

matteobruni committed:
Rename custom.md to pull_request_template.md

matteobruni committed:
Merge branch 'master' into r-1.18.11

matteobruni committed:
build: webpack configs for presets and shapes have now a common file

code-factor committed:
[CodeFactor] Apply fixes to commit d234d48

[ci skip] [skip ci]

matteobruni committed:
refactor(core): splitted particle links drawing from particle drawing

matteobruni committed:
docs: updated options for the animations of size and opacity

matteobruni committed:
chore(release): published new version

 - ng-particles@2.1.11
 - inferno-particles@1.1.11
 - jquery-particles@1.18.11
 - preact-particles@1.18.11
 - react-tsparticles@1.18.11
 - svelte-particles@1.1.11
 - particles.vue@2.1.11
 - particles.vue3@1.1.11
 - tsparticles-editor@1.18.11
 - tsparticles@1.18.11
 - ng-particles-demo@1.3.11
 - editor-demo@1.3.11
 - inferno-particles-demo@1.1.11
 - jquery-particles-demo@1.3.11
 - tsparticles-demo@1.3.11
 - preact-particles-demo@1.3.11
 - react-tsparticles-demo@1.3.11
 - svelte-demo@1.1.11
 - particles.vue-demo@1.3.11
 - particles.vue3-demo@1.1.11
 - tsparticles-preset-60fps@1.3.11
 - tsparticles-preset-background-mask@1.3.11
 - tsparticles-preset-basic@1.3.11
 - tsparticles-preset-big-circles@1.3.11
 - tsparticles-preset-bouncing@1.3.11
 - tsparticles-preset-fire@1.3.11
 - tsparticles-preset-font-awesome@1.3.11
 - tsparticles-preset-fountain@1.3.11
 - tsparticles-preset-snow@1.3.11
 - tsparticles-preset-stars@1.3.11
 - tsparticles-shape-bubble@1.3.11
 - tsparticles-shape-heart@1.3.11
 - tsparticles-shape-multiline-text@1.3.11
 - tsparticles-shape-spiral@1.3.11
 - cra-template-particles-typescript@1.1.11
 - cra-template-particles@1.1.11

matteobruni committed:
Merge pull request #1002 from matteobruni/r-1.18.11

1.18.11

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
Merge branch 'master' into dev

matteobruni committed:
fix: fixed removeAt with groups

matteobruni committed:
build: fixed webpack configs

matteobruni committed:
build: fixed webpack configs

matteobruni committed:
build: removed unused code

matteobruni committed:
feat: emitters now can have a spawn color that is also animatable

matteobruni committed:
feat: mouse trail can now be stopped when mouse stops

matteobruni committed:
build: fixed broken build

matteobruni committed:
Merge branch 'dev' into 188-add-orbit-feature

matteobruni committed:
chore(demo): improved sea anemone demo

matteobruni committed:
feat: prepared spin move options
feat: noise have a clamp option to clamp speed inside [-1, 1] or not
chore(demo): improved sea anemone demo

matteobruni committed:
Merge branch 'dev' into 188-add-orbit-feature

matteobruni committed:
chore(demo): improved sea anemone demo

Commit:
feat: animation, random for orbit angle and redrawn orbit

Commit:
Merge branch '188-add-orbit-feature' of https://github.com/thecodingdog/tsparticles into 188-add-orbit-feature

Commit:
refactor: orbit

Commit:
refactor: orbit

matteobruni committed:
Merge pull request #986 from thecodingdog/188-add-orbit-feature

feat: add static orbit

matteobruni committed:
build: updated rush config

matteobruni committed:
build: restored yarn and lerna, rush was not working well in this project (for now)

matteobruni committed:
build: removed some presets, they will be deprecated and replaced by others more standalone

matteobruni committed:
chore(release): published new version

 - ng-particles@2.2.0-alpha.0
 - inferno-particles@1.2.0-alpha.0
 - jquery-particles@1.19.0-alpha.0
 - preact-particles@1.19.0-alpha.0
 - react-tsparticles@1.19.0-alpha.0
 - svelte-particles@1.2.0-alpha.0
 - particles.vue@2.2.0-alpha.0
 - particles.vue3@1.2.0-alpha.0
 - tsparticles-editor@1.19.0-alpha.0
 - tsparticles@1.19.0-alpha.0
 - ng-particles-demo@1.4.0-alpha.0
 - editor-demo@1.4.0-alpha.0
 - inferno-particles-demo@1.2.0-alpha.0
 - jquery-particles-demo@1.4.0-alpha.0
 - tsparticles-demo@1.4.0-alpha.0
 - preact-particles-demo@1.4.0-alpha.0
 - react-tsparticles-demo@1.4.0-alpha.0
 - svelte-demo@1.2.0-alpha.0
 - particles.vue-demo@1.4.0-alpha.0
 - particles.vue3-demo@1.2.0-alpha.0
 - tsparticles-preset-big-circles@1.4.0-alpha.0
 - tsparticles-preset-fire@1.4.0-alpha.0
 - tsparticles-preset-fountain@1.4.0-alpha.0
 - tsparticles-preset-basic@1.4.0-alpha.0
 - tsparticles-preset-snow@1.4.0-alpha.0
 - tsparticles-preset-stars@1.4.0-alpha.0
 - tsparticles-shape-bubble@1.4.0-alpha.0
 - tsparticles-shape-heart@1.4.0-alpha.0
 - tsparticles-shape-multiline-text@1.4.0-alpha.0
 - tsparticles-shape-spiral@1.4.0-alpha.0
 - cra-template-particles-typescript@1.2.0-alpha.0
 - cra-template-particles@1.2.0-alpha.0

matteobruni committed:
feat: added noise generator management like shape drawers and plugins
feat: added sea anemone preset
fix: fixed main demo
refactor: renamed basic preset to links
